structs.go

```diff
@@ -839,14 +839,23 @@ type GuildScheduledEventParams struct {
 // GuildScheduledEventParams to handle the special case of a null value
 // for channel_id field
 func (p *GuildScheduledEventParams) MarshalJSON() ([]byte, error) {
```

Comment:
Should we use `GuildScheduledEventParams` instead of `*GuildScheduledEventParams`, since we don't change anything in the base instance